src/NServiceBus.Core/obsoletes.cs

```diff
@@ -2254,6 +2255,24 @@ public void ApplyTo(Type componentType, IComponentConfig config)
         {
             throw new NotImplementedException();
         }
+
+        [ObsoleteEx(
+          RemoveInVersion = "7.0",
+          TreatAsErrorFromVersion = "6.0",
+          Message = "Set(string key, object value)")]
```

Comment:
yeah, good catch, need to use ReplacementTypeOrMember :+1: 

I'm on the fence about the obsolete message. I doubt that anyone is using it (because it doesn't nearly work as you'd expect it).
> and pass the settings holder to the object needing the settings. Extract the settings and set the state accordingly.

that was already needed before. The SetProperty did nothing else than just a Set with the object full name + propertyname. It doesn't change anything on the access side.